### PR TITLE
feat: Add refocus prop to prevent refocus after close

### PR DESCRIPTION
**Pull request by gavmck**
We have some instances where we do not want to refocus the last focused element when closing the modal as this causes some unexpected scroll jumping.

I have added a prop of `refocus` which can be set to `false` to prevent refocus of the last element focused.

Thanks! Let me know if you need any changes.

### CODE REVIEW

**Comment by LouisAugry** [comment]
Nice catch ! ;) Can you implement the test ?

File: src/ngx-smart-modal/src/services/ngx-smart-modal.service.ts [hunk 1]
```diff
@@ -383,7 +383,9 @@ export class NgxSmartModalService {
       modal.markForCheck();
       modal.onCloseFinished.emit(modal);
       modal.onAnyCloseEventFinished.emit(modal);
-      this.lastElementFocused.focus();
+      if (modal.refocus) {
+        this.lastElementFocused.focus();
+      }
```
Comment (LouisAugry) [inline src/ngx-smart-modal/src/services/ngx-smart-modal.service.ts]: Line 258, you forgot 
```js
if (typeof options.refocus === 'boolean ') { componentRef.instance.refocus = options.refocus; }
```

Comment (gavmck) [reply src/ngx-smart-modal/src/services/ngx-smart-modal.service.ts]: Thanks, fixed!

**Comment by gavmck** [comment]
> Nice catch ! ;) Can you implement the test ?

@LouisAugry I've just tried running the tests, but I get hit by a ton of Typescript errors before I've made any changes. 

Do they work for you?

```
ERROR in src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts(197,54): error TS2345: Argument of type 'false' is not assignable to parameter of type 'Element'.
src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts(220,54): error TS2345: Argument of type '{ getBoundingClientRect: () => { left: number; top: number; width: number; height: number; }; }' is not assignable to parameter of type 'Element'.
  Type '{ getBoundingClientRect: () => { left: number; top: number; width: number; height: number; }; }' is missing the following properties from type 'Element': assignedSlot, attributes, classList, className, and 122 more.
src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts(221,57): error TS2322: Type 'number' is not assignable to type 'string'.
src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts(221,72): error TS2322: Type 'number' is not assignable to type 'string'.
src/ngx-smart-modal/tests/services/ngx-smart-modal-stack.service.spec.ts(49,55): error TS2345: Argument of type '{ modal: { layerPosition: number; visible: boolean; }; }[]' is not assignable to parameter of type 'ModalInstance[]'.
  Property 'id' is missing in type '{ modal: { layerPosition: number; visible: boolean; }; }' but required in type 'ModalInstance'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(14,4): error TS2554: Expected 2 arguments, but got 1.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(218,43): error TS2345: Argument of type '{ getData: () => string; }' is not assignable to parameter of type 'NgxSmartModalComponent'.
  Type '{ getData: () => string; }' is missing the following properties from type 'NgxSmartModalComponent': closable, escapable, dismissable, identifier, and 58 more.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(244,43): error TS2345: Argument of type 'false' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(269,48): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(277,43): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(286,43): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(295,43): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(305,58): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'ModalInstance[]'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(316,62): error TS2345: Argument of type '"fake"' is not assignable to parameter of type 'NgxSmartModalComponent'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(329,56): error TS2322: Type 'string' is not assignable to type 'ModalInstance'.
src/ngx-smart-modal/tests/services/ngx-smart-modal.service.spec.ts(330,72): error TS2322: Type 'string' is not assignable to type 'NgxSmartModalComponent'.
```

**Comment by LouisAugry** [comment]
Yes they works perfectly ! Look at the travis-ci job 😉

**Comment by DanWebb** [comment]
@LouisAugry Tests added!

**Comment by gavmck** [comment]
@LouisAugry Do you know when this might get released? Getting chased as the QA team keep picking it up :)

**Comment by maximelafarie** [comment]
@gavmck @DanWebb I'm releasing it now. Good job guys! 🤘

**Comment by gavmck** [comment]
Awesome, thank you!

**Comment by DanWebb** [comment]
Thanks @maximelafarie!

**Comment by DanWebb** [comment]
@maximelafarie Any chance we could get an NPM release for this 🙏

**Comment by maximelafarie** [comment]
@DanWebb I'm releasing it now. 😉Sorry for the waiting but I had to check our planned releases on our roadmap. 🖖

**Comment by maximelafarie** [comment]
@gavmck @DanWebb the 7.4.0 has been released! Thank you for your patience and for using our lib and contribute to it! Cheers! 🖖

**Pull request merged**